### DISCUSSION TITLE
WebUI admin/user mode?

### DISCUSSION THREAD

**Discussion by ALfuhrmann** [arendst/Tasmota#14987] [Q&A] [answered]
Is there a possibility to log in as admin when the WebUI runs in user mode (WebServer 1)?

So that normal users are able to toggle relais, but not to screw up the configuration?

**Comment by sfromis** [comment, marked as answer]
No. You can "screw up" the config via MQTT, including temporarily disabling user mode.

**Reply by ALfuhrmann** [reply]
I'm not talking about an attack, I just want to keep the use of the WebUI as admin tool (e.g. TasmotaAdmin) while having a way to allow non-admin web use for the family.

**Reply by barbudor** [reply]
As of today, the WebServer is designed to work in either USER or ADMIN mode (or OFF of course)
My understanding is that you would like someone loging as a USER (no password I would assume) to see the server in USER mode and someone logging as ADMIN (with password) to see the pages as ADMIN.
The main problem is that currently the Tasmota webserver do not manage any session context that would be the base to be able to manage such thing.

As there is no will to add more code to expand in that direction, I would suggest to manage that on a Home Automation system.

**Reply by sfromis** [reply]
I was not talking about an attack either, just what you **can** do for admin when you run your Tasmota device in user mode.

**Reply by ALfuhrmann** [reply]
> My understanding is that you would like someone loging as a USER (no password I would assume) to see the server in USER mode and someone logging as ADMIN (with password) to see the pages as ADMIN.

Exactly. Maybe using a different URL or an "admin" button.

> The main problem is that currently the Tasmota webserver do not manage any session context that would be the base to be able to manage such thing.

OK, that's what I was getting at. So I can only set the WebUI as a USER and administer the device via MQTT.
And let my users switch on the lights using Domoticz.

Thank you!

Edit: formatting

**Reply by sfromis** [reply]
You can also use MQTT to temporarily switch to `WebServer 1`, user mode, if you feel more comfortable using the web UI.

**Comment by Zumili** [comment]
A bit late, but if you just want to prevent your family from messing up stuff by pressing the wrong buttons and if you are willing to use "scripting" instead of "rules" you can simulate some "kind" of USER mode by adding the following CSS code to your **>W** section of the script.

`<style>form button{display:none}</style>`

This removes the lower standard buttons. If you have added buttons via the script, they will not get affected. You can still reach the different page with a direct link like so:

`http://ipofpi/cs?`

**Reply by ALfuhrmann** [reply]
A neat little hack!

Thank you!

**Reply by Elektrofix-OL** [reply]
Is it possible to switch off individual buttons (e.g. settings).
I didn't find anything in the Q&A and the documentation.

**Reply by sfromis** [reply]
Assuming that you talk about the Tasmota front page, you do not have features for this being a customizable "dashboard" for daily operation, the intention is that regular control happens in your backend setup, not the rudimentary admin/test web UI.

**Reply by Elektrofix-OL** [reply]
Yes, I'm talking about the Tasmota home page. with the code <style>form button{display:none}</style>
all buttons disappear.
I want to keep the "information" and "restart" button.

The background is that I would like to enable an old woman to read her meter on her cell phone without having to always climb down the steps to the meter. If there are problems, she can send me a screenshot of the info page or do a reset and I don't have to go to the old lady.

I hope to get a tip for the code to be able to show the buttons individually.

**Reply by barbudor** [reply]
If you have Scripting-Language to do the HTML inject above, you can also inject your own buttons to replace those you have made disapear

**Reply by Zumili** [reply]
Have a look at the "Inspector" tool of your browser, you will see, that the buttons are inside a form, these forms have IDs (but0, but1, ...)
So you can hide e.g. Information-button with the following CSS:
`<style>#but4 button{display:none}</style>`

Checked two devices and in my case the IDs are as follows:
```
but0: Restart
but1: ---
but2: ---
but3: Configuration
but4: Information
but5: Firmware Upgrade
but6: Consoles
```
To hide all _except_ **Information** and **Restart**:
```
<style>#but3 button{display:none}</style>
<style>#but5 button{display:none}</style>
<style>#but6 button{display:none}</style>
```
Check if these IDs are correct by using the Inspector tool of your browser because I cannot guarantee that these IDs are always the same.
Again, keep in mind, that Tasmota WebUI is more for admins than for users.

**Reply by barbudor** [reply]
Excellent 👍

**Comment by Elektrofix-OL** [comment]
Many thanks for your response.
My programming knowledge isn't that fresh anymore. Back then, C64, TurboPascal and access2000 were the state of the art, so I sometimes miss the syntex. I just finished it.
Later I will publish the project unter https://github.com/Elektrofix-OL so that other people can use it.
![image](https://user-images.githubusercontent.com/39242936/222093703-b1bc84f6-6ca9-466f-8986-e0124b75499c.png)
